pkg/obfuscator/replacement_reporter.go

```diff
@@ -12,6 +12,9 @@ type ReplacementReporter interface {
 	// ReportReplacement will add a replacement along with its original string to the report.
 	// If there is an existing value that does not match the given replacement, it will panic as this very likely denotes a bug.
 	ReportReplacement(original string, replacement string)
+
+	// HasReplacement returns the previously used replacement if already set.
+	HasReplacement(original string) string
```

Comment:
so it's a `GetReplacement`? please also document the behaviour if it doesn't exist and add tests.